frontend/src/pages/ActivityReport/index.js

```diff
@@ -207,8 +259,27 @@ function ActivityReport({
           && (report.calculatedStatus === REPORT_STATUSES.DRAFT
             || report.calculatedStatus === REPORT_STATUSES.NEEDS_ACTION);
         updateAdditionalData({ recipients, collaborators, availableApprovers });
+
+        let shouldUpdateFromNetwork = true;
+
+        if (formData && savedToStorage) {
```

Comment:
Just add a comment here explaining that this is checking if the last local storage save is older than the report last updated at.